Ncsdk 4043 release notes

ru-fu opened the pull request:
Please check and comment with additions and changes.

PDF printout: 
[nRF Connect SDK v1.2.0 Release Notes — nRF Connect SDK 1.1.pdf](https://github.com/NordicPlayground/fw-nrfconnect-nrf/files/4203985/nRF.Connect.SDK.v1.2.0.Release.Notes.nRF.Connect.SDK.1.1.pdf)



Check here once all input for your team (Change log and Known issues) has been provided:
- [x] Code owners
- [x] Hussar - Protocol
- [x] Hussar - LowLevel
- [x] IoT
- [x] Aurora
- [x] Crypto
- [x] Thingy:91
- [x] Desktop
- [x] Dragoon
- [x] Test
- [x] Project management

Approve when all content for your team is included correctly.

ru-fu commented:
@MonikaLieLarsen

MonikaLieLarsen commented on doc/release-notes-1.2.0.rst:
```diff
@@ -0,0 +1,482 @@
+|NCS| delivers reference software and supporting libraries for developing low-power wireless applications with Nordic Semiconductor products.
+It includes the MCUboot and the Zephyr RTOS open source projects, which are continuously integrated and re-distributed with the SDK.
+
+|NCS| v1.1.0 supports product development with the nRF9160 Cellular IoT device.
```
v1.2.0

MonikaLieLarsen commented on doc/release-notes-1.2.0.rst:
```diff
@@ -0,0 +1,482 @@
+Highlights
+**********
+
+* ???
```
Experimental support for the nRF5340 device

MonikaLieLarsen commented on doc/release-notes-1.2.0.rst:
```diff
@@ -0,0 +1,482 @@
+See :ref:`Getting the nRF Connect SDK code <cloning_the_repositories_win>` for more information.
+
+
+Supported modem firmware
```
Perhaps "Supported modem firmware for Cellular IoT applications" ?

ru-fu replied:
A bit too long for the heading, but I added an introduction - ok?

Supported modem firmware
************************

This version of the |NCS| supports the following modem firmware for cellular IoT applications:

* mfw_nrf9160_1.2.0

shantha-14 replied:
Modem version is not corect. mfw_nrf9160_1.1.1

ru-fu replied:
OK, updated.

umapraseeda commented on doc/release-notes-1.2.0.rst:
```diff
@@ -0,0 +1,482 @@
+
+  * :ref:`cloud_client` - shows how to connect to and communicate with a cloud service using the generic :ref:`cloud_api_readme`.
+  * :ref:`https_client` - shows how to provision a TLS certificate and connect to an HTTPS server.
+  * :ref:`serial_lte_modem` - demostrates sending AT commands between a host and a client device.
```
```suggestion
  * :ref:`serial_lte_modem` - demonstrates sending AT commands between a host and a client device.
```

frkv commented on doc/release-notes-1.2.0.rst:
```diff
@@ -0,0 +1,490 @@
+Crypto
+======
+
+???
```
Crypto additions

* Added low-level cryptographic test suite using NIST, RFCs and custom test vectors.
* Added nrf_cc310_mbedcrypto/nrf_cc310_platform library version 0.9.2:
  * Fixes power-efficiency issues. 
  * Library adds experimental use of CryptoCell interrupt instead of busy-waits.

hw_cc310 driver addition
* Added support for CryptoCell interrupt in driver.

Raane commented on doc/release-notes-1.2.0.rst:
```diff
@@ -0,0 +1,490 @@
+  * :ref:`modem_key_mgmt` - provides functions to provision security credentials to the nRF9160 modem.
+    The library replaces the ``nrf_inbuilt_key`` APIs from the :ref:`nrfxlib:bsdlib`.
+  * :ref:`lib_zzhc` - implements the self-registration functionality that is required to connect to the China Telecom network.
+  * :ref:`supl_client` - ???
```
Externally hosted SUPL client library is now supported. Implements AGPS data downloading from SUPL server.

Raane commented on doc/release-notes-1.2.0.rst:
```diff
@@ -0,0 +1,490 @@
+
+* :ref:`gps_with_supl_support_sample`:
+
+  * ???
```
GPS sample supports the new SUPL client library if enabled.

shantha-14 commented on doc/release-notes-1.2.0.rst:
```diff
@@ -0,0 +1,490 @@
+     - `GNU Arm Embedded Toolchain`_
+
+
+As IDE, we recommend to use |SES| (Nordic Edition) version 4.44a. ??? correct?
```
yes

thst-nordic commented:
master is ready for rebase to fix documentation build.

evenl commented on doc/release-notes-1.2.0.rst:
```diff
@@ -0,0 +1,499 @@
+* nRF91 fails to receive large packets (over 4000 bytes).
+* nrf_connect fails if called immediately after initialization of the device.
+  A delay of 1000 ms is required for this to work as intended.
+
```
* The GPS sample will stop working if the if SUPL client library support is enabled and it fails to resolve the SUPL hostname.
As a workaround can a few second delay, using k_sleep(), be inserted after the printf on line 294 in main.c.

oyvindronningstad commented on doc/release-notes-1.2.0.rst:
```diff
@@ -0,0 +1,499 @@
+Bootloader
+----------
+
+* Public keys are not revoked when subsequent keys are used.
```
This issue has been fixed.

oyvindronningstad commented on doc/release-notes-1.2.0.rst:
```diff
@@ -0,0 +1,499 @@
+----------
+
+* Public keys are not revoked when subsequent keys are used.
+* The bootloader does not work properly on nRF51.
```
and nRF53

oyvindronningstad commented on doc/release-notes-1.2.0.rst:
```diff
@@ -0,0 +1,499 @@
+* Public keys are not revoked when subsequent keys are used.
+* The bootloader does not work properly on nRF51.
+* Building and programming the immutable bootloader (see :ref:`ug_bootloader`) is not supported in SEGGER Embedded Studio.
+* The immutable bootlader can only be used with the following boards:
```
typo bootlader -> bootloader

oyvindronningstad replied:
Should we merge this point with the one above about nRF51 (and now nRF53)?

ru-fu commented on doc/release-notes-1.2.0.rst:
```diff
@@ -0,0 +1,490 @@
+
+* Added the following libraries:
+
+  * ??? fprotect?
```
:ref:`fprotect_readme` - can be used to protect flash areas from writing.
    This library was extracted from of the :ref:`bootloader` sample.

Correct?

oyvindronningstad replied:
Exactly, thanks.

kapi-no commented:
In case you want to see the full diff between the current master and v1.1.0 release, here is a link to it:
https://github.com/NordicPlayground/fw-nrfconnect-nrf/compare/v1.1.0...master

jtguggedal commented on doc/release-notes-1.2.0.rst:
```diff
@@ -0,0 +1,502 @@
+
+* :ref:`lib_nrf_cloud`:
+
+  * Removed the button/switch pairing method from nRF Cloud FSM. ??? what is FSM?
```
FSM is the finite state machine. Can just replace it with `library`, I think. One can no longer use button/switch association method with nRF Cloud lib.

jtguggedal commented on doc/release-notes-1.2.0.rst:
```diff
@@ -0,0 +1,502 @@
+  * Added high/low threshold configuration command handling for sensor data. ??? what? Added handling for commands that configure the threshold? Added command handling for threshold configurations?
+  * Modified the command format to match the format that is used by nRF Cloud.
+  * Implemented support for receiving modem AT commands from the cloud and returning the modem's response.
+  * Implemented interval commands from the cloud to the device. ??? what are interval commands? Missing in the documentation.
```
The interval is how often sensor data is sent to nRF Cloud.
@jayteemo has the details on this and the threshold configuration

jayteemo replied:
Interval, in seconds, at which the device will send sensor data to the cloud. Disabled by setting to 0 or null.

Light Sensor:
`{"appId":"LIGHT", "messageType":"CFG_SET","data": {"interval": 20}}`

Env (temperature, humidity, air pressure, and air quality) Sensor:
`{"appId":"ENV", "messageType":"CFG_SET","data": {"interval": 20}}`

ru-fu replied:
Updated to:
  * Added functionality to configure the interval at which sensor data is sent to the cloud.
    This makes it possible to change the poll/send interval for environmental and light sensors from the terminal card in nRF Cloud.

jtguggedal commented on doc/release-notes-1.2.0.rst:
```diff
@@ -0,0 +1,502 @@
+
+* :ref:`asset_tracker`:
+
+  * Added high/low threshold configuration command handling for sensor data. ??? what? Added handling for commands that configure the threshold? Added command handling for threshold configurations?
```
Thresholds can be set so that only sensor data below or above certain thresholds is sent.
@jayteemo Is the support for this in the cloud UI in place? Could you add examples of commands to set thresholds, intervals and send AT commands somewhere publicly available?

jayteemo replied:
@ru-fu that is correct.
Example JSON command:
```
{
  "appId": "TEMP",
  "messageType": "CFG_SET",
  "data": {
    "enable": true,
    "thresh_lo": 20,
    "thresh_hi": 50
  }
} 
```
The above JSON can be sent to the device through the device's `Terminal` card on nrfcloud.com.
The device would then only report temperatures less than 20 or greater than 50.

jayteemo replied:
example AT command sent from the `Terminal` card on nrfcloud.com: 
`{"appId":"MODEM", "messageType":"CMD", "data":"AT+CFUN?\r\n"}`
AT response is then displayed in the `Terminal` card.

What is the desired _public place_ for such examples?

hakonfam commented on doc/release-notes-1.2.0.rst:
```diff
@@ -0,0 +1,502 @@
+
+  * Added the configuration option :option:`CONFIG_DFU_TARGET_MCUBOOT_SAVE_PROGRESS`, which uses Zephyr's :ref:`zephyr:settings` subsystem.
+    When this option is enabled, the write progress of an MCUboot style upgrade is stored, so that the progress is retained when the device reboots.
+  * Fixed a bug where :cpp:func:`dfu_target_done` printed ``unable to deinitialize dfu resource`` when no target was initialized. ??? did it actually print it or return an error?
```
It returned an error, and the error is fixed

sigvartmh replied:
Missed this but yes an error was outputted in the log and an error value was returned.

hakonfam commented on doc/release-notes-1.2.0.rst:
```diff
@@ -0,0 +1,502 @@
+Build system
+============
+
+* Fixed a bug where a user-defined HEX file was not included in the merge operation. ??? what is the context here? Partition Manager?
```
The context is providing static configuration to the partition manager

hakonfam replied:
Yes! Thanks!

oyvindronningstad commented on doc/release-notes-1.2.0.rst:
```diff
@@ -0,0 +1,502 @@
+* :ref:`bootloader`:
+
+  * Moved the provisioning data (slot sizes/addresses and public keys) to one-time programmable (OTP) memory for nRF9160 devices.
+  * Implemented invalidation of public keys.
```
@hakonfam should we mention that the bootloader uses NCS_SAMPLES_DEFAULTS?

hakonfam replied:
I see no reason for that

bjorn-tore-taraldsen commented on doc/release-notes-1.2.0.rst:
```diff
@@ -0,0 +1,502 @@
+
+|NCS| v1.2.0 supports product development with the nRF9160 Cellular IoT device.
+It contains reference applications, sample source code, and libraries for Bluetooth Low Energy devices in the nRF52 Series, though product development on these devices is not currently supported with the |NCS|.
+??? What do we say about nRF5340?
```
Replace with the following text. This will also include nRF53 in the scope: 
"It contains reference applications, sample source code, and libraries for developing low power wireless applications with nRF52 & nRF53 series devices, though support for these devices is incomplete and not recommended for production."

rugeGerritsen commented on doc/release-notes-1.2.0.rst:
```diff
@@ -0,0 +1,502 @@
+* PCA10059 (nRF52840 Dongle)
+* PCA10040 (nRF52 Development Kit)
+* PCA10028 (nRF51 Development Kit)
+
```
What about support for nrf52811 and nrf52833?

shantha-14 replied:
PCA10028  should be removed. nRF52811 and 833 is not tested. I am not comfortable in adding it to the supported boards. Agree @bjorn-tore-taraldsen

bjorn-tore-taraldsen replied:
The question is - What do we mean by "Supported boards" here? 
Is it DK's that we have available as a target for our samples or the one we have used during development/ testing?

shantha-14 replied:
I would claim supported if and only if we have tested. Otherwise how would we know whether it works ? @carlescufi  How  is the claim done in the zephyr world ?

carlescufi replied:
There is no claim upstream. All support is "best effort", so not comparable. I would stick to the boards we have actually tested on.

ru-fu replied:
@bjorn-tore-taraldsen @shantha-14 Define text to explain what supported means.

ru-fu replied:
Leave list as it is now.

rugeGerritsen commented on doc/release-notes-1.2.0.rst:
```diff
@@ -0,0 +1,502 @@
+==================
+
+* Updated the :ref:`nrfxlib:ble_controller` to v. ??? what version? The changelog seems to be messed up.
+  For details, see the :ref:`nrfxlib:ble_controller_changelog`. ??? update link
```
Suggested text:
```suggestion
Updated the nRF BLE Controller libraries and removed their version numbers: 
 * Added preliminary support for the S140 variant with the nRF5340 device. The BLE controller for nRF5340 supports the same feature set as its nRF52 Series counterpart.
 * The library now must be linked together with MPSL. Some APIs are moved to MPSL.
 * Data Length Extensions is now a configurable feature
 * Fixed an issue where an assert could occur when receiving a packet with a CRC error after performing a data length procedure on Coded PHY.

For details, see the :ref:`nrfxlib:_ble_controller_changelog`.
```

rugeGerritsen commented on doc/release-notes-1.2.0.rst:
```diff
@@ -0,0 +1,502 @@
+Multi-Protocol Service Layer (MPSL)
+===================================
+
+* Added MPSL v0.5.0-1.prealpha, a library of common services for single and multi-protocol implementations.
```
```suggestion
Updated the MPSL libraries and removed their version numbers:
 * Added a library version with preliminary support for the nRF5340 device. The feature set is the same as in the MPSL library for nRF52.
 
For details, see the :ref:`nrfxlib:_mpsl_changelog:`.
```

rugeGerritsen commented on doc/release-notes-1.2.0.rst:
```diff
@@ -0,0 +1,502 @@
+
+Subsystems
+==========
+
```
MPSL has been added as a new subsystem. I'm not sure if it should be mentioned here
```suggestion
Multi-Protocol Service Layer (MPSL)
---------------------------------------
 * MPSL has been added as a new subsystem. It integrates the MPSL into the NCS environment

```

bjorn-tore-taraldsen replied:
Agree this should be included. I've also mention this in the Highlight.

rugeGerritsen commented on doc/release-notes-1.2.0.rst:
```diff
@@ -0,0 +1,502 @@
+Bluetooth Low Energy
+--------------------
+
+* :option:`CONFIG_BT_HCI_TX_STACK_SIZE` must be set to 1536 when selecting :option:`CONFIG_BT_LL_NRFXLIB`.
```
The known BLE controller issues from last release are now resolved. These can therefore be removed.
At the moment there are no new known issues.

ru-fu replied:
Removed the following issues:

```
* :option:`CONFIG_BT_HCI_TX_STACK_SIZE` must be set to 1536 when selecting :option:`CONFIG_BT_LL_NRFXLIB`.
* The :ref:`nrfxlib:ble_controller` 0.3.0-3.prealpha might assert when receiving a packet with an CRC error on LE Coded PHY after performing a DLE procedure where RX Octets is changed to a value above 140.
* :option:`CONFIG_SYSTEM_WORKQUEUE_STACK_SIZE` must be set to 2048 when selecting :option:`CONFIG_BT_LL_NRFXLIB` on :ref:`central_uart` and :ref:`central_bas`.
* :option:`CONFIG_NFCT_IRQ_PRIORITY` must be set to 5 or less when selecting :option:`CONFIG_BT_LL_NRFXLIB` on :ref:`peripheral_hids_keyboard`.
* When selecting :option:`CONFIG_BT_LL_NRFXLIB`:
  If a directed high duty cycle advertiser times out, the application might have to wait a short time before starting a new connectable advertiser.
  Otherwise, starting the advertiser will fail.
```
Correct? Did I miss any? @rugeGerritsen

kapi-no commented on doc/release-notes-1.2.0.rst:
```diff
@@ -0,0 +1,502 @@
+
+Bluetooth Low Energy
+--------------------
+
```
* The :ref:\`peripheral_hids_keyboard\` sample cannot be used with the :ref:\`nrfxlib:ble_controller\` because NFC subsystem cannot work together with the controller library. The library uses MPSL Clock driver, which does not provide API for asynchronous clock operation and NFC requires it to work correctly.
* When the :ref:\`peripheral_hids_mouse\` sample is used with the Zephyr controller, directed advertising does not timeout and the regular advertising cannot be started.
* The :ref:\`bluetooth_central_hids\` sample cannot connect to the peripheral that uses directed advertising.

kapi-no commented on doc/release-notes-1.2.0.rst:
```diff
@@ -0,0 +1,502 @@
+
+  * Bluetooth Low Energy:
+
+    * :ref:`ble_llpm` - added ??? not mentioned in release notes
```
@ryanjh, can you provide some input for your sample in the Bluetooth subsystem section?

rugeGerritsen replied:
I would propose: Added a sample to showcase the proprietary Low Latency Packet Mode (LLPM) extension

kapi-no commented on doc/release-notes-1.2.0.rst:
```diff
@@ -0,0 +1,502 @@
+  * Bluetooth Low Energy:
+
+    * :ref:`ble_llpm` - added ??? not mentioned in release notes
+    * :ref:`bluetooth_mesh_light` - added ??? not mentioned in release notes
```
@trond-snekvik, please cover missing parts for Bluetooth mesh

ru-fu replied:
We decided to leave Mesh out of the release notes, so no input needed after all.
I'll remove the mentions here.

jhn-nordic commented on doc/release-notes-1.2.0.rst:
```diff
@@ -0,0 +1,502 @@
+
+nRF9160
+=======
+
```
* In the time window between a PSM request and a successful response, sending LTE packets can cause the send() function in bsdlib to hang for long periods. (from seconds to tens of minutes) Behavior depends on cellular network connection.  

@jtguggedal  or @evenl please update if you have a better suggestion to text.

jhn-nordic replied:
send() on a socket.

PSM = Power saving mode

jhn-nordic replied:
The originial description was not correct. Here is a new proposal.  "nrf_send() can be blocking for several minutes even when socket is configured for non blocking operation" @jtguggedal @evenl please comment.

jtguggedal replied:
I think it's correct to say that a `send()` call can block even though it's configured to be non-blocking.

jhn-nordic replied:
Agree.

jtguggedal replied:
But I don't have detailed knowledge of bsdlib's inner workings, so can't really say what is the "most correct" phrasing here

ru-fu replied:
Updated to:
* The :cpp:func:`nrf_send` function in the :ref:`nrfxlib:bsdlib` might be blocking for several minutes, even when the socket is configured for non-blocking operation.
  The behavior depends on the cellular network connection.

sigvartmh commented on doc/release-notes-1.2.0.rst:
```diff
@@ -0,0 +1,502 @@
+  As a workaround, call :cpp:func:`fota_download_start` again with the same arguments.
+  A fix for this issue is available in commit `38625ba7 <https://github.com/NordicPlayground/fw-nrfconnect-nrf/commit/38625ba775adda3cdc7dbf516eeb3943c7403227>`_.
+* If the last fragment of a :ref:`lib_fota_download` is received but is corrupted, or if the last write is unsuccessful, the library emits an error event as expected.
+  However, it also emits an apply/request update event, even though the downloaded data is invalid.
```
This section can be removed.

tejlmand commented on doc/release-notes-1.2.0.rst:
```diff
@@ -0,0 +1,502 @@
+     - `GNU Arm Embedded Toolchain`_
+
+
+As IDE, we recommend to use |SES| (Nordic Edition) version 4.44a.
```
```suggestion
As IDE, we recommend to use |SES| (Nordic Edition) version 4.42a.
```

nordic-krch commented on doc/release-notes-1.2.0.rst:
```diff
@@ -0,0 +1,502 @@
+* In the BSD library, the GNSS sockets implementation is experimental.
+
+
+nrfx v1.8.1
```
nrfx is in version  v2.1.0. @kl-cruz  can you confirm that these known issues are no longer valid?

nika-nordic replied:
I can confirm these issues have been fixed in nrfx v2.1.0

kapi-no replied:
Speaking of known issues in nrfx. Could you please mention incorrect pin definition in MDK? The number of P1 port pins is wrong for nRF5340. I created a known issue for NFC that manifests itself because of it:

https://github.com/NordicPlayground/fw-nrfconnect-nrf/pull/1854/files#r378233002

I think it is worth mentioning somewhere. What's your take on it?

nika-nordic replied:
@kapi-no I think it's worth mentioning here. It would go like this:
* nRF5340:
  Pins P1.12 to P1.16 are unavailable due to incorrect pin number definition in MDK

nika-nordic replied:
Sorry, in fact it is P1.12 to P1.15, so here is corrected version:
* nRF5340:
  Pins P1.12 to P1.15 are unavailable due to incorrect pin number definition in MDK

ru-fu replied:
Removed nrfx section and added an "MDK (part of nrfx)" section.

bjorn-tore-taraldsen commented on doc/release-notes-1.2.0.rst:
```diff
@@ -0,0 +1,550 @@
+??? Where does this go?
+=======================
+
+Settings: Reduced the default partition size for settings storage.
```
Where: New section under "Subsystems" named "Settings" or "Setting storage" 

Change the text to: 
"Default partition size for the settings storage reduced from 24kB (0x6000) to 8kB (0x2000). This gives more flash space for the application."

barsok commented:
The LL Team are not reviewers so we cannot edit or request changes or tick the LL box. Our input is ready, we have left comments.

kapi-no commented on doc/release-notes-1.2.0.rst:
```diff
@@ -0,0 +1,502 @@
+---
+
+???
+
```
yes

oyvindronningstad commented on doc/release-notes-1.2.0.rst:
```diff
@@ -0,0 +1,550 @@
+* On nRF51 devices, Bluetooth LE samples that use GPIO might crash when buttons are pressed frequently.
+  In such case, the GPIO ISR introduces latency that violates real-time requirements of the Radio ISR.
+  nRF51 is more sensitive to this issue than nRF52 (faster core).
+
```
Not sure about the best placement for this:

> BLE cannot be used in a non-secure app, i.e. the nrf5340_dk_nrf5340_cpuappns board. Use the nrf5340_dk_nrf5340_cpuapp board instead.

cc @bjorn-tore-taraldsen @rugeGerritsen

bjorn-tore-taraldsen replied:
@grochu Ref. our discussion aboyt Sec/ non-se. This is the conclusion from Øyvind. Any comment

grochu replied:
We might want to add a note or update somehow the nRF53 User Guide.

ru-fu replied:
Too late for doc updates now - do we expect this to be a persistent problem, or will we fix it?

Added to the BLE section now.

bjorn-tore-taraldsen commented on doc/release-notes-1.2.0.rst:
```diff
@@ -0,0 +1,550 @@
+
+Highlights
+**********
+
```
New highlight section (replacing what's there)
-	Added support for the new nRF5340 device 
-	New Multi-Protocol Service Layer (MPSL) included in nrfxlib (link?)
-	NFC supporting the new TNEP protocol  
-	New nRF9160 samples and libraries
-	General updates and bugfixes

bjorn-tore-taraldsen commented on doc/release-notes-1.2.0.rst:
```diff
@@ -0,0 +1,550 @@
+  * :ref:`supl_client` - integrates the externally hosted SUPL client library.
+    This library implements A-GPS data downloading from a SUPL server.
+
+
```
New section named "nRF5340" with some extra details on introducing this new chip.
-   nrfx supporting nRF5340
-   Added support for the :ref:`nRF5340 PDK board (PCA10095)<nrf5340_dk_nrf5340>` with board targets nrf5340_dk_nrf5340_cpunet and nrf5340_dk_nrf5340_cpuapp
- Demonstrating a dual core solution with Bluetooth LE Controller running on the network core and Bluetooth host and application running on the application core. Running HCI over openAMP.
        > Both the Nordic nRF Bluetooth LE Controller and Zephyr Bluetooth LE Controller is ported running on the network core (nrf5340_dk_nrf5340_cpunet)
       > Demonstrating Bluetooth LE samples both nRF Connect Samples and Zephyr Samples running on application core (nrf5340_dk_nrf5340_cpuapp)
-	Added RadioTest Sample (ported from legacy nRF5 SDK) running on the network core (nrf5340_dk_nrf5340_cpunet)
-	Added NFC support

nvlsianpu commented:
@ru-fu Below note on changes taken from upsterm mcuboot since NCS 1.1.0
I don't know where to put this, so please redact and take this into the patch.

```
MCUboot adds support for encrypted images using
ECIES with secp256r1 as an Elliptic Curve alternative to RSA-OAEP. A
new swap method was added which allows for upgrades without using a
scratch partition. There are also lots of bug fixes, extra simulator
testing coverage and some imgtool updates.

### About this release

- TLVs were updated to use 16-bit lengths (from previous 8). This
  should work with no changes for little-endian targets, but will
  break compatibility with big-endian targets.
- A benchmark framework was added to Zephyr
- imgtool was updated to correctly detect trailer overruns by image.
- Encrypted image TLVs can be saved in swap metadata during a swap
  upgrade instead of the plain AES key.
- imgtool can dump private keys in C format (getpriv command), which
  can be added as decryption keys. Optionally can remove superfluous
  fields from the ASN1 by passing it `--minimal`.
- Lots of other smaller bugs fixes.
- Added downgrade prevention feature (available when the overwrite-based
  image update strategy is used)

### Known issues (not important for the nRF at all)

- TLV size change breaks compatibility with big-endian targets.
```

nvlsianpu commented:
^^ To known issues:
```
mcuboot recovery feature using USB interface doesn't work
```

grochu commented on doc/release-notes-1.2.0.rst:
```diff
@@ -0,0 +1,550 @@
+    This sample was ported from the nRF5 SDK.
+
+* Extended the :ref:`nfc_tag_reader` sample with parsing and printing of the Type 4 Tag content.
+
```
Moved the NFC Platform implementation to the _fw-nrfconnect-nrf_ repo (related docs in nrfxlib: `:ref:` nfc_integration_notes)

grochu replied:
All NFC samples: enabled by default:

stack protection,
assertions
logging

shantha-14 commented on doc/release-notes-1.2.0.rst:
```diff
@@ -0,0 +1,550 @@
+
+  * :ref:`radio_test` - demonstrates how to configure the radio in a specific mode and then test its performance.
+    This sample was ported from the nRF5 SDK.
+
```
missing entry for ble mesh examples @trond-snekvik

ru-fu replied:
I discussed with @bjorn-tore-taraldsen and @MonikaLieLarsen yesterday and the decision was to leave Mesh out of the release notes.

bjorn-tore-taraldsen replied:
We will leave Mesh out of the scope for what's mention for this release.

shantha-14 commented on doc/release-notes-1.2.0.rst:
```diff
@@ -0,0 +1,550 @@
+
+* Enabled UART flow control in the :ref:`peripheral_uart` sample to avoid data loss.
+
+* Fixed the :ref:`ble_throughput` sample, which did not achieve the expected throughput measurements.
```
should we state this, still with Zephyr LL we do not achieve the expected throughput

kapi-no replied:
@joerchan, how should we correct it? I added this line based on your recent changes to the Throughput sample.

ru-fu replied:
@kapi-no to check with @joerchan

kapi-no replied:
Please replace it with this one:

Changed the :ref:\`ble_throughput\` sample to prevent it from running BLE scanning and advertising in parallel. The feature to establish connection in both master and slave role at the same time is not supported by the Zephyr BLE Host.

shantha-14 commented on doc/release-notes-1.2.0.rst:
```diff
@@ -0,0 +1,550 @@
+  There is an incorrect number of pins defined in the MDK files, and the pins required for :ref:`st25r3911b_nfc_readme` cannot be configured properly.
+* NFC tag samples are unstable when exhaustively tested (performing many repeated read and/or write operations).
+  NFC tag data might be corrupted.
+
```
NFC poller samples do not work for nRF53 targets

kapi-no replied:
Its already there, but phrased differently :)

grochu commented on doc/release-notes-1.2.0.rst:
```diff
@@ -0,0 +1,550 @@
+
+* Enabled UART flow control in the :ref:`peripheral_uart` sample to avoid data loss.
+
+* Fixed the :ref:`ble_throughput` sample, which did not achieve the expected throughput measurements.
```
All BLE samples: enabled by default:
- stack protection,
- assertions
- logging

grochu replied:
All BLE samples: modified to use synchronous `bt_enable()` function.

grochu replied:
Discovery Manager: fixed an issue, where service or characteristic allocation failed, but the returned pointer was not checked before accessing data it was pointing to.

grochu commented on doc/release-notes-1.2.0.rst:
```diff
@@ -0,0 +1,550 @@
+
+* Enabled UART flow control in the :ref:`peripheral_uart` sample to avoid data loss.
+
+* Fixed the :ref:`ble_throughput` sample, which did not achieve the expected throughput measurements.
```
HID Service: use `bt_gatt_is_subscribed()` instead of an internal implementation in the HID service

joerchan replied:
> HID Service: use `bt_gatt_is_subscribed()` instead of an internal implementation in the HID service

This doesn't affect the user I think. Only an internal change in the service.

ru-fu replied:
Not added for now.

bjorn-tore-taraldsen commented:
Ref. comment from @nvlsianpu regarding mcuBoot.
We should only list new mcuBoot feature included: - Support for encrypted Images, - New Swap method removing need for a scratch partition, - bug fixes.
and refer to mcuBoot for details. 
Here (https://github.com/JuulLabs-OSS/mcuboot/blob/master/docs/release-notes.md#version-150) is a super set of what has been included. We are missing one feature "ed25519 signature verification can't be build relaying on bundled tinycrypt based sha-512 implementation" from this list.

shantha-14 commented on doc/release-notes-1.2.0.rst:
```diff
@@ -0,0 +1,550 @@
+     - Version
+     - Download link
+   * - SEGGER J-Link
+     - V6.50b
```
this is not correct....6.60e

ru-fu commented:
@bjorn-tore-taraldsen @nvlsianpu 
Added the following section for MCUboot:


```
MCUboot
=======

* Updated to include new features from upstream:

  * Support for encrypted images
  * New swap method that removes the need for a scratch partition
  * Bug fixes

  See the :ref:`MCUboot release notes <https://github.com/JuulLabs-OSS/mcuboot/blob/master/docs/release-notes.md#version-150>`_ for more information.
  Note that not all features from v1.5.0 are included.
```

OK?

ru-fu commented on doc/release-notes-1.2.0.rst:
```diff
@@ -0,0 +1,550 @@
+  The behavior depends on the cellular network connection.
+* The :ref:`gps_with_supl_support_sample` sample stops working if :ref:`supl_client` support is enabled, but the SUPL host name cannot be resolved.
+  As a workaround, insert a delay (``k_sleep()``) of a few seconds after the ``printf`` on line 294 in :file:`main.c`.
+* Deprecation warning: The nrf_inbuilt_key API in the :ref:`nrfxlib:bsdlib` will be removed in a future release.
```
I think this has been removed? @Raane

ru-fu replied:
Remove.

ru-fu commented on doc/release-notes-1.2.0.rst:
```diff
@@ -0,0 +1,550 @@
+  * Other:
+
+    * :ref:`radio_test` - added
+    * :ref:`timeslot_sample` - added ??? not mentioned in release notes
```
Who is responsible for this sample, and in what category should it be listed? Common samples or MPSL subsystem?
@rugeGerritsen

rugeGerritsen replied:
@ru-fu . @joerchan and me are responsible for this sample. It demonstrates functionality provided by MPSL. I think "Other" is the correct category for now

rugeGerritsen replied:
"This sample demonstrates how to use :ref:`nrfxlib:mpsl` and basic MPSL Timeslot functionality."

ru-fu replied:
Agreed to put under MPSL subsystem.

ru-fu commented:
Updated - please check (the PDF in the first post is updated as well).

Also, please check the unresolved conversations.

rugeGerritsen commented on doc/release-notes-1.2.0.rst:
```diff
@@ -0,0 +1,586 @@
+* The central samples (:ref:`central_uart`, :ref:`bluetooth_central_hids`) do not support any pairing methods with MITM protection.
+* On some operating systems, the nrf_desktop application is unable to reconnect to a host.
+* central_uart: A too long 212-byte string cannot be handled when entered to the console to send to peripheral_uart.
+* On nRF51 devices, Bluetooth LE samples that use GPIO might crash when buttons are pressed frequently.
```
Should BLE on nrf51 be removed, as it is no longer one of the supported boards?

ru-fu replied:
Remove

ru-fu commented on doc/release-notes-1.2.0.rst:
```diff
@@ -0,0 +1,586 @@
+
+* :ref:`lib_dfu_target`:
+
+  * Added the configuration option :option:`CONFIG_DFU_TARGET_MCUBOOT_SAVE_PROGRESS`, which uses Zephyr's :ref:`zephyr:settings` subsystem.
```
@hakonfam Should this be removed?

ru-fu replied:
Keep.

nvlsianpu commented on doc/release-notes-1.2.0.rst:
```diff
@@ -0,0 +1,586 @@
+
+* Updated to include new features from upstream:
+
+  * Support for encrypted images
```
some implementation primitives were changed, but encrypted image are there since a while. Remove this line

nvlsianpu commented on doc/release-notes-1.2.0.rst:
```diff
@@ -0,0 +1,586 @@
+
+  * Support for encrypted images
+  * New swap method that removes the need for a scratch partition
+  * Bug fixes
```
Add point: `Added downgrade prevention feature (available when the overwrite-based
  image update strategy is used)`

jtguggedal commented on doc/release-notes-1.2.0.rst:
```diff
@@ -0,0 +1,586 @@
+
+  * :ref:`at_cmd_readme`
+  * ``lte_link_control``
+
```
The AT host library has changes default line ending from `CR` to `LF` to support sending SMS using the AT host

ru-fu commented on doc/release-notes-1.2.0.rst:
```diff
@@ -0,0 +1,586 @@
+NFC
+---
+
+* The NFC poller samples cannot be run on the nRF5340 PDK.
```
NFC poller and NFC reader samples (:ref:)

kapi-no replied:
:ref:\`nfc_tnep_poller\`
:ref:\`nfc_tag_reader\`

jhn-nordic commented on doc/release-notes-1.2.0.rst:
```diff
@@ -0,0 +1,586 @@
+  As a workaround, insert a delay (``k_sleep()``) of a few seconds after the ``printf`` on line 294 in :file:`main.c`.
+* Deprecation warning: The nrf_inbuilt_key API in the :ref:`nrfxlib:bsdlib` will be removed in a future release.
+  A replacement library that wraps the AT commands for ``AT%CMNG`` will be available in the |NCS|.
+* The :ref:`asset_tracker` sample might show up to 2.5 mA current consumption in idle mode with ``CONFIG_POWER_OPTIMIZATION_ENABLE=y``.
```
@jorgenmk is this still valid?

grochu commented on doc/release-notes-1.2.0.rst:
```diff
@@ -0,0 +1,590 @@
+  However, security is not enabled by default in the Zephyr sample.
+* The central samples (:ref:`central_uart`, :ref:`bluetooth_central_hids`) do not support any pairing methods with MITM protection.
+* On some operating systems, the nrf_desktop application is unable to reconnect to a host.
+* central_uart: A too long 212-byte string cannot be handled when entered to the console to send to peripheral_uart.
```
This has been fixed.

grochu commented on doc/release-notes-1.2.0.rst:
```diff
@@ -0,0 +1,590 @@
+  NFC requires this API to work correctly.
+* When the :ref:`peripheral_hids_mouse` sample is used with the Zephyr Bluetooth LE Controller, directed advertising does not time out and the regular advertising cannot be started.
+* The :ref:`bluetooth_central_hids` sample cannot connect to a peripheral that uses directed advertising.
+* Bluetooth Low Energy peripheral samples are unstable in some conditions (when pairing and bonding are performed and then disconnections/re-connections happen).
```
I think since the last NCS release, most of the stability issues in BLE samples have been fixed. Most of them were caused by incorrect default configuration. @shantha-14 can we remove this point?

ru-fu replied:
Removed.

carlescufi commented on doc/release-notes-1.2.0.rst:
```diff
@@ -0,0 +1,590 @@
+
+* Updated to v2.1.0.
+  For details, see the `changelog <https://github.com/NordicSemiconductor/nrfx/blob/v2.1.0/CHANGELOG.md>`_.
+
```
@ru-fu In order to keep this simple I will refer to the upstream revisions:

```
Zephyr
======

This nRF Connect SDK released is based on Zephyr 2.1.99 (and in particular on Zephyr SHA 40175fd3bd), which is revision sitting
in between the upstream Zephyr 2.1 and 2.2 releases. 

To see a comprehensive list of changes introduced since nRF Connect 1.1.0 you can use the following Git command:

.. code-block:: console
   git log 7d7fed0d2b..40175fd3bd
```

carlescufi commented:
@ru-fu we are still missing hussar protocol and PMT?

ru-fu commented:
> 
> 
> @ru-fu we are still missing hussar protocol and PMT?

No, I think we're good to merge. :)
